### PR TITLE
[swift] to support application/octet-stream for file upload

### LINKED ISSUE #15129: [REQ] Feature Request Description
### Is your feature request related to a problem? Please describe.

Yes, I would need to support "application/octet-stream" in the Swift generator.

## Describe the solution you'd like

1. In `URLSessionRequestBuilder.execute()`, check if `contentType.hasPrefix("application/octet-stream")` and initialize `OctetStreamEncoding()` 
2. Create a new `OctetStreamEncoding` class
3. Duplicate `FormURLEncoding` content and replace "application/x-www-form-urlencoded" by "application/octet-stream" to `OctetStreamEncoding`

## Describe alternatives you've considered

Override `URLSessionRequestBuilder.execute()` but it will duplicate too many lines

## Additional context

I need to be able to upload a file into a server accepting only `content-type: "application/octet-stream"`

### PR DESCRIPTION
**Pull request by aniruddhJo**
To fix #15129

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache [hunk 1]
```diff
@@ -618,6 +620,24 @@ private class FormURLEncoding: ParameterEncoding {
     }
 }
 
+private class OctetStreamEncoding: ParameterEncoding {
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache:623]: thanks for the PR. can you please add a line or 2 explaining what this function does?

**Comment by wing328** [comment]
cc @antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @jirikuncar (2021/01) @ph4r5h4d (2021/04) @lwj5 (2023/04)

**Comment by lwj5** [comment]
Erm. This is swift...

**Comment by lwj5** [comment]
Swift	@jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11)

**Comment by wing328** [comment]
oops.. sorry. thanks for the correction

File: modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache [hunk 2]
```diff
@@ -128,6 +128,8 @@ private var credentialStore = SynchronizedDictionary<Int, URLCredential>()
                 encoding = FormDataEncoding(contentTypeForFormPart: contentTypeForFormPart(fileURL:))
             } else if contentType.hasPrefix("application/x-www-form-urlencoded") {
                 encoding = FormURLEncoding()
+            } else if contentType.hasPrefix("application/octet-stream"){
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache:131]: @aniruddhJo Hi, thanks for creating this PR.
Could you please also add support for `application/octet-stream` in alamofire please?
Thanks

Comment (wing328) [reply modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache:131]: created a ticket to track this: https://github.com/OpenAPITools/openapi-generator/issues/16300

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator/pull/16299

**Pull request closed without merging**